src/main/webapp/app/iris/base-chatbot/chat-status-bar/chat-status-bar.component.ts

```diff
@@ -28,8 +33,8 @@ export class ChatStatusBarComponent implements OnChanges {
 
     ngOnChanges() {
         // Lower case state for scss classes, avoid function calling in template
-        this.stages?.forEach((stage) => (stage.lowerCaseState = stage.state?.toLowerCase()));
-        const firstUnfinished = this.stages?.find((stage) => !this.isStageFinished(stage));
+        this.stages()?.forEach((stage) => (stage.lowerCaseState = stage.state?.toLowerCase()));
+        const firstUnfinished = this.stages()?.find((stage) => !this.isStageFinished(stage));
```

Comment:
if you plan to move the other variables (open etc.) also to signals I would suggest to move the complete ngOnChanges logic into an effect which runs everytime when `stages` changes

Reply:
That's an excellent point! I implemented an effect that runs every time stages change.